spec/lib/message_bus_spec.rb

```diff
@@ -216,6 +196,50 @@
     @bus.backlog("/chuck").map { |i| i.data }.to_a.must_equal ['foo']
   end
 
+  it "can be turned off" do
+    @bus.off
+
+    @bus.off?.must_equal true
+
+    @bus.publish("/chuck", "norris")
+
+    @bus.backlog("/chuck").to_a.must_equal []
+  end
+
+  it "can be turned off only for subscriptions" do
+    @bus.off(disable_publish: false)
+
+    @bus.off?.must_equal true
+
+    data = nil
+
+    @bus.subscribe("/chuck") do |msg|
+      data = msg.data
+    end
+
+    @bus.publish("/chuck", "norris")
+
+    @bus.backlog("/chuck").map(&:data).to_a.must_equal ["norris"]
+
+    sleep 2
```

Comment:
this is tricky, need to figure out how to do this without the sleep... maybe we tag as a memory only test and then we can control the time better. 

This is not an easy feature to test, for sure.